test-framework/junit5/src/main/java/io/quarkus/test/junit/AppMakerHelper.java

```diff
@@ -225,12 +227,22 @@ public CuratedApplication makeCuratedApplication(Class<?> requiredTestClass, Str
             }
 
             addToBuilderIfConditionMet.accept(appClassLocation);
-            final Path appResourcesLocation = PathTestHelper.getResourcesForClassesDirOrNull(appClassLocation, "main");
-            addToBuilderIfConditionMet.accept(appResourcesLocation);
+
+            testAppModel = BootstrapAppModelFactory.newInstance()
+                    .setTest(true)
+                    .setEnableClasspathCache(true)
+                    .setProjectRoot(projectRoot)
+                    .resolveAppModel()
+                    .getApplicationModel();
+
+            for (var rootDir : testAppModel.getAppArtifact().getContentTree().getRoots()) {
+                addToBuilderIfConditionMet.accept(rootDir);
+            }
```

Comment:
Here we resolve the `ApplicationModel` to get the proper application output directories. Without this, the first test added by @holly-cummins will fail. Because the path resolution logic for `tests-jar` files resolves the main JARs corresponding to those `tests-jar` as the "application root", which is not enough in this case because the `HelloResource` will not be visible (indexed) when a test app is bootstrap.